docs/CONFIGURATION.md

```diff
@@ -0,0 +1,188 @@
+# Configuring the Mountpoint for Amazon S3 CSI Driver
+
+See [the Mountpoint documentation](https://github.com/awslabs/mountpoint-s3/blob/main/doc/CONFIGURATION.md) for
+Mountpoint specific configuration.
+
+## AWS Credentials
+
+The Mountpoint CSI Driver can be configured to ingest credentials via two approaches: globally for the entire 
+Kubernetes cluster, or using credentials assigned to pods.
+
+### Driver Level Credentials
+
+By setting driver-level credentials, the whole cluster uses the same set of credentials. 
+
+Using this configuration, the credentials that are used are set at installation time, either using Service
+Accounts, or using K8s secrets. 
+
+See [install.md](./install.md) for more details on this approach.
+
+**TODO** - Migrate driver level setup here.
+
+### Driver Level Credentials with IRSA
+
+Configuring [IAM Roles for Service Accounts](https://docs.aws.amazon.com/eks/latest/userguide/iam-roles-for-service-accounts.html) (IRSA)
+is the recommended way to set up the CSI Driver if you want to use Driver Level credentials.
+
+```mermaid
+graph LR;
+    CSI[CSI Driver]
+
+    P["`Application Pod
+    *Driver IAM Credentials*`"]
+
+    SA_D[Service Account - Driver]
+    SA_P[Service Account - Pod]
+
+    IAM_D[IAM Credentials - Driver]
+    IAM_P[IAM Credentials - Pod]
+
+    PV[Persistent Volume]
+    PVC[PVC]
+
+    P --> PVC
+
+    PVC --> PV
+
+    PV --> CSI
+
+    CSI --> SA_D
+    P --> SA_P
+
+    SA_D --> IAM_D
+    SA_P --> IAM_P
+```
+
+### Driver Level Credentials with K8s Secrets
+
+The Mountpoint CSI Driver also supports sourcing long term AWS credentials from k8s secrets, though this is not 
+recommended.
```

Comment:
I'd recommend rephrasing this more like:

```suggestion
Where IAM Roles for Service Accounts (IRSA) isn't a viable option, Mountpoint CSI Driver also supports sourcing static AWS credentials from K8s secrets.

> [!WARNING]
> We do not recommend using long-term AWS credentials and K8s secrets. Instead, we recommend using short-term credentials with IRSA.
```

Reply:
I'm changing this to
> We do not recommend using long-term AWS credentials. Instead, we recommend using short-term credentials with IRSA.

and removing the bit about k8s secrets, as I don't think we should be against recommending their usage in general